packages/flutter/lib/src/widgets/localized_resources.dart

```diff
@@ -0,0 +1,354 @@
+/// classes that contain `Intl.message()` calls. In this case that would just be the
+/// `MyLocalizedResources` class.
+///
+/// One could choose another approach for loading localized resources and looking them up while
+/// still conforming to the structure of this example.
+class LocalizedResources extends StatefulWidget {
+  LocalizedResources({
+    Key key,
+    @required this.locale,
+    this.delegate,
+    this.child
+  }) : super(key: key) {
+    assert(locale != null);
+  }
+
+  /// The resources returned by [LocalizedResources.of] will be specific to this locale.
+  final Locale locale;
+
+  /// This delegate is responsible for loading and caching resources for `locale`.
+  ///
+  /// Typically defined with [DefaultLocalizedResourcesDelegate].
+  final LocalizedResourcesDelegate delegate;
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  /// Returns the localized resources for the widget tree that corresponds to
+  /// [BuildContext] `context`.
+  ///
+  /// This method is typically combined with [LocalizedResourcesData.for] to defined
+  /// a static method that looks up a collection of related related resources. For
+  /// example Flutter looks up Material resources with a method defined like this:
+  ///
+  /// ```dart
+  /// static LocalizedMaterialResources of(BuildContext context) {
+  ///  return LocalizedResources.of(context)
+  ///    .resourcesFor<LocalizedMaterialResources>(LocalizedMaterialResources);
+  /// }
+  /// ```
+  static LocalizedResourcesData of(BuildContext context) {
+    final _LocalizedResourcesScope scope = context.inheritFromWidgetOfExactType(_LocalizedResourcesScope);
+    return scope?.localizedResourcesState;
+  }
+
+  @override
+  _LocalizedResourcesState createState() => new _LocalizedResourcesState();
+}
+
+class _LocalizedResourcesState extends State<LocalizedResources> implements LocalizedResourcesData {
+  final GlobalKey _localizedResourcesScopeKey = new GlobalKey();
+
+  @override
+  Locale get locale => _locale;
+  Locale _locale;
+
+  @override
+  void initState() {
+    super.initState();
+    load(widget.locale);
+  }
+
+  void load(Locale locale) {
+    if (widget.delegate == null) {
+      setState(() {
+        _locale = locale;
+      });
+      return;
+    }
+
+    widget.delegate.load(locale).then((_) {
+      setState(() {
+        _locale = locale;
+      });
+      final InheritedElement scopeElement = _localizedResourcesScopeKey.currentContext;
+      scopeElement?.dispatchDidChangeDependencies();
+    });
+  }
+
+  @override
+  T resourcesFor<T>(Type type) => widget.delegate.resourcesFor<T>(_locale, type);
+
+  @override
+  Widget build(BuildContext context) {
+    return new _LocalizedResourcesScope(
+      key: _localizedResourcesScopeKey,
+      locale: widget.locale,
+      localizedResourcesState: this,
+      child: _locale != null ? widget.child : new Container(),
```

Comment:
Do we document what we do when `_locale` is `null`? Also, is this what we _want_ to do. Replacing the child with an arbitrary `Container` doesn't feel like a graceful back-off strategy.